dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java

```diff
@@ -907,4 +919,66 @@ public Group createDefaultReadGroup(Context context, Collection collection, Stri
         return role;
     }
 
+    @Override
+    public List<Collection> findAuthorizedCollectionsInSOLR(String q, Context context, Community community,
+        int offset, int limit) throws SQLException, SearchServiceException {
+
+        List<Collection> collections = new ArrayList<Collection>();
+        StringBuilder query = new StringBuilder();
+        DiscoverQuery discoverQuery = new DiscoverQuery();
+        discoverQuery.setDSpaceObjectFilter(IndexableCollection.TYPE);
+        discoverQuery.setStart(offset);
+        discoverQuery.setMaxResults(limit);
+        DiscoverResult resp = resultSolrQuery(context, query, discoverQuery,community, q);
+        for (IndexableObject solrCollections : resp.getIndexableObjects()) {
+            Collection c = ((IndexableCollection) solrCollections).getIndexedObject();
+            collections.add(c);
+        }
+        return collections;
+    }
+
+    @Override
+    public int countAuthorizedCollectionsInSOLR(String q, Context context, Community community)
+        throws SQLException, SearchServiceException {
+
+        StringBuilder query = new StringBuilder();
+        DiscoverQuery discoverQuery = new DiscoverQuery();
```

Comment:
As we only need the total results for this one can you call on:
`discoverQuery.setMaxResults(0);`

Because the default here is 10 & we don't need to retrieve these objects from the database if we need the total count.